### ISSUE TITLE
Some confusion over enum

### ISSUE THREAD

**Issue by leastbad** [jasl/activeentity#13]
I don't know if this is an issue, or possibly just an opportunity to create more documentation for the `enum` module.

While I think I have come to understand the Active Entity enum implementation, my experiments suggest that its behaviour is different enough from `ActiveRecord::Enum` that it's *almost* harmful to use the AR documentation for this feature.

Unlike AR `enum`, AE `enum` only accepts one parameter, and that parameter must be a Hash. This makes most of the examples at the top of https://github.com/jasl/activeentity/blob/master/lib/active_entity/enum.rb incorrect; for example, there are no scopes created and there is no concept of `where` querying with AE. It supports suffixes and prefixes, but no other options.

AR `enum` is described as "an enum attribute where the values map to integers in the database, but can be queried by name" but that is just not how the AE implementation works.

```rb
class Example < ActiveEntity::Base
  attribute :status, :integer, default: 0
  enum status: [:active, :inactive]
end

example = Example.new
example.attributes # => {"status"=>"active"}
```

Really, what we have is **not an Integer**, but a String that can be set with an Integer. I don't know if AE tracks the type cast but the important thing is that if I go to work with the `attributes` it shows up as a String... meaning that logically, I should define `status` as a String attribute when using Active Entity. However, that doesn't end well:

```rb
class Example < ActiveEntity::Base
  attribute :status, :string, default: 10
  enum status: {active: 10, inactive: 20}
end

example = Example.new
example.attributes # => {"status"=>"active"}
example.inactive! # => {"status"=>nil} <- I'm cheating a bit because AllFutures supports update!/save!
example.status = 20 # => {"status"=>"inactive"}
example.status = :active # => {"status"=>"active"}
```

So, to recap: AE `enum` looks like and behaves as a String, but must be declared with Integer type or else the dynamic methods are not created properly.

I actually think that this is **fine** now that I have mapped it out, but that we have a responsibility to document these quirks. In the StimulusReflex project, our first step in solving any problem is to try and ask if it is something that should just be fully explained in the docs.

I am writing docs for AllFutures and would be happy to share anything useful from that work upstream to Active Entity.

**Comment by jasl** [comment]
Basically I remove all DB related codes so no scope is by design.

I must confess update document and tests are big lacking for AE, since I knew there are people rely on this gem, I must thinking these carefully.

if you can point out which part made you confusing or trouble, we can start there, help is abosulutely welcome.

**Comment by leastbad** [comment]
As I said: I actually think that this might be able to be fixed with some additions to `README.md`.

It has some quirks, but that is to be expected. Honestly, my friends and I have been incredibly impressed with how much work you have done on this library.

I will prepare a PR that addresses `enum` as it applies to AE.

**Comment by jasl** [comment]
> As I said: I actually think that this might be able to be fixed with some additions to `README.md`.
> 
> It has some quirks, but that is to be expected. Honestly, my friends and I have been incredibly impressed with how much work you have done on this library.
> 
> I will prepare a PR that addresses `enum` as it applies to AE.

Thank you!
